applications/StructuralMechanicsApplication/custom_elements/truss_element_3D2N.cpp

```diff
@@ -300,8 +300,14 @@ void TrussElement3D2N::CalculateRightHandSide(
   KRATOS_TRY
   rRightHandSideVector = ZeroVector(msLocalSize);
 
-  BoundedVector<double, msLocalSize> internal_forces = ZeroVector(msLocalSize);
-  this->UpdateInternalForces(internal_forces);
+  BoundedVector<double,msLocalSize> internal_forces =
+    this->GetConstitutiveLawTrialResponse(rCurrentProcessInfo,false);
+
+  BoundedMatrix<double, msLocalSize, msLocalSize> transformation_matrix =
+      ZeroMatrix(msLocalSize, msLocalSize);
+  this->CreateTransformationMatrix(transformation_matrix);
+
+  internal_forces = prod(transformation_matrix, internal_forces);
   noalias(rRightHandSideVector) -= internal_forces;

```

Comment:
why not 
`noalias(rRightHandSideVector) -= prod(transformation_matrix, internal_forces);`

Reply:
i thought splitting it up makes it easier to understand

Reply:
In principle this is a good idea but I am not sure if this creates overhead ...
can you do `noalias(internal_forces)` ?

It is called quite often since it is in CalcRHS, so it should be efficient

@loumalouomega any opinion on this

Reply:
Can you try this Klaus?

Reply:
done